# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoaderBase.cpp
# Commit: 68b476066248
# Full Hash: 68b476066248deb9b80b0be44bebb991fdc94a6d
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-12-21 21:21:23
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement InitModuleLoader methods for WorkerScriptLoader and WorkerGlobalScope; r=jonco
#   
#   This implements a method to initialize the moduleLoader for workers. This will initialize only once, for all worker types (module and classic).
#   
#   Depends on D147324
# ==============================================================================

diff -r c94a9dc60dff -r 68b476066248 js/loader/ModuleLoaderBase.cpp
--- a/js/loader/ModuleLoaderBase.cpp	Tue Dec 20 20:56:11 2022 +0000
+++ b/js/loader/ModuleLoaderBase.cpp	Tue Dec 20 20:56:12 2022 +0000
@@ -983,7 +983,9 @@
   MOZ_ASSERT(mFetchingModules.IsEmpty());
 
   for (const auto& entry : mFetchedModules) {
-    entry.GetData()->Shutdown();
+    if (entry.GetData()) {
+      entry.GetData()->Shutdown();
+    }
   }
 
   mFetchedModules.Clear();
